torchtune/training/precision.py

```diff
@@ -160,6 +160,8 @@ def validate_expected_param_dtype(
         ValueError: If any parameter has a different dtype than `dtype`.
     """
     for name, param in named_params:
+        if "vocab_pruning_module" in name: 
```

Comment:
might be more generalizable to add an optional argument to ignore certain named params. As it is it is quite specific